src/bugreporter/odemis_bugreporter.py

```diff
@@ -0,0 +1,520 @@
+        self.known_user_parser = ConfigParser.ConfigParser(dict_type=OrderedDict)
+        conf_path = os.path.join(os.path.expanduser(u"~"), '.config', 'odemis', 'odemis-bugreporter.config')
+        with open(conf_path, 'r+') as f:
+            self.known_user_parser.readfp(f)
+        if 'users' not in self.known_user_parser.sections():
+            self.known_user_parser.add_section('users')
+
+    def on_name_key_down(self, evt):
+        """
+        If the pressed key is the backspace, delete suggestion and allow user to type new key.
+        """
+        if evt.GetKeyCode() == wx.WXK_BACK:
+            self.make_suggestion = False
+        else:
+            self.make_suggestion = True
+        evt.Skip()
+
+    def on_name_text(self, _):
+        """
+        Suggest a username from the configuration file.
+        """
+        if self.make_suggestion:
+            full_text = self.name_ctrl.GetValue()
+            if not full_text:
+                return
+
+            # Get typed text from input field (don't include suggestion)
+            if self.name_ctrl.GetSelection()[0] != self.name_ctrl.GetSelection()[1]:
+                typed = full_text[:len(full_text) - self.name_ctrl.GetSelection()[0] + 1]
+            else:
+                typed = full_text
+
+            # Suggest name from configuration file and select suggested part
+            for name in self.known_user_parser.options('users'):
+                if name.startswith(typed):
+                    wx.CallAfter(self.name_ctrl.ChangeValue, name)
+                    break
+                else:
+                    wx.CallAfter(self.name_ctrl.ChangeValue, typed)
+            wx.CallAfter(self.name_ctrl.SetSelection, len(typed), -1)
+
+    def on_name_focus(self, _):
+        """
+        Suggest email address for username.
+        """
+        name = self.name_ctrl.GetValue()
+        if name in self.known_user_parser.options('users'):
+            self.email_ctrl.ChangeValue(self.known_user_parser.get('users', name))
+
+    def on_report_btn(self, evt):
+        """
+        Disable all widgets, send ticket, save name and email in configuration file.
+        """
+        self.wait_lbl.SetLabel("Compressing files...")
+        self.Layout()
+
+        name = self.name_ctrl.GetValue()
+        email = self.email_ctrl.GetValue()
+        summary = self.summary_ctrl.GetValue()
+        description = self.description_ctrl.GetValue()
+        
+        if not name or not email or not summary or not description:
+            dlg = wx.MessageDialog(self, 'Please fill in all the fields.', '', wx.OK)
+            val = dlg.ShowModal()
+            dlg.Show()
+            if val == wx.ID_OK:
+                dlg.Destroy()
+                return
+
+        for widget in self.panel.GetChildren():
+            widget.Enable(False)
+            self.wait_lbl.Enable(True)
+        # Add user to top of config file, truncate file if it contains too many users.
```

Comment:
Maybe move all this suggestion-related code to a separate function, like "store_user_info()", to clarify a little this function.
Also, are any character supported? If not, are they automatically filtered out?